### DISCUSSION TITLE
darkMode selector strategy in 4.0.0-alpha.16

### DISCUSSION THREAD

**Discussion by sk-one-it** [tailwindlabs/tailwindcss#13863] [Help] [answered]
Guys, I have been playing around with the alpha version of tailwind and I was trying to implement dark mode. The default strategy of setting dark theme based on prefers-color-scheme works well. And I am now trying to implement the manual strategy (giving user the ability to change theme).

You can see in the below image that I tried applying dark class to html element but to no avail. Since I am trying this in React, apart from window.load, I have tried useLayoutEffect as well to add the dark class to the html element.

![image](https://github.com/tailwindlabs/tailwindcss/assets/165276471/aa6a3cdd-8cf7-4626-9ba7-0badfd6fd5b7)

I have been told that the alpha version of tailwind does not support config file. Can someone please tell me how I can go about the dark theme feature in the alpha version of tailwind?

**Comment by wongjn** [comment, marked as answer]
Only the media query version of dark mode is supported in v4 at the moment. As per the [release blog post](https://tailwindcss.com/blog/tailwindcss-v4-alpha#roadmap-to-v4-0):

> - **Support for other dark modes** — right now we only support dark mode using media queries, and still need to reimplement the selector and variant strategies.

Also, https://github.com/tailwindlabs/tailwindcss/issues/13772#issuecomment-2142855927

**Comment by Bishwas-py** [comment]
Try this: https://github.com/tailwindlabs/tailwindcss/issues/13772#issuecomment-2416740649
